src/Core/SonarLintNotifications/INotificationSettingStorage.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SonarLint.VisualStudio.Core.SonarLintNotifications
+{
+    public enum StorageLocation
+    {
+        User,
+        VSInstance,
+        Solution
+    }
+
+    public interface INotificationSettingStorage
+    {
+        void SaveNotificationSetting(INotificationSetting setting, StorageLocation storageLocation);
```

Comment:
Naming: `Save` and `Load`, or `Set` and `Get` (combining `Save` and `Get` looks a bit odd).